File: pkg/dev_compiler/test/codegen_expected/corelib/reg_exp_start_end_test.js
dart_library.library('corelib/reg_exp_start_end_test', null, /* Imports */[
  'dart_sdk',
  'expect'
], function load__reg_exp_start_end_test(exports, dart_sdk, expect) {
  'use strict';
  const core = dart_sdk.core;
  const dart = dart_sdk.dart;
  const dartx = dart_sdk.dartx;
  const expect$ = expect.expect;
  const reg_exp_start_end_test = Object.create(null);
  let VoidTodynamic = () => (VoidTodynamic = dart.constFn(dart.definiteFunctionType(dart.dynamic, [])))();
  reg_exp_start_end_test.main = function() {
    let matches = core.RegExp.new("(a(b)((c|de)+))").allMatches("abcde abcde abcde");
    let it = matches[dartx.iterator];
    let start = 0;
    let end = 5;
    while (dart.test(it.moveNext())) {
      let match = it.current;
      expect$.Expect.equals(start, match.start);
      expect$.Expect.equals(end, match.end);
      start = start + 6;
      end = end + 6;
    }
  };
  dart.fn(reg_exp_start_end_test.main, VoidTodynamic());
  // Exports:
  exports.reg_exp_start_end_test = reg_exp_start_end_test;
});

File: tests/corelib/reg_exp_start_end_test.dart
// Copyright (c) 2011, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

import "package:expect/expect.dart";

main() {
  var matches = new RegExp("(a(b)((c|de)+))").allMatches("abcde abcde abcde");
  var it = matches.iterator;
  int start = 0;
  int end = 5;
  while (it.moveNext()) {
    Match match = it.current;
    Expect.equals(start, match.start);
    Expect.equals(end, match.end);
    start += 6;
    end += 6;
  }
}
